modules/sfp_snov.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_snov
+# Purpose:      Spiderfoot plugin to search Snov.IO API for emails 
+#               associated to target domain
+#
+# Author:      Krishnasis Mandal <krishnasis@hotmail.com>
+#
+# Created:     16/05/2020
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+from netaddr import IPNetwork
+import urllib.request, urllib.parse, urllib.error
+import json
+
+class sfp_snov(SpiderFootPlugin):
+    """Snov:Investigate,Passive:Reputation Systems:client_id,client_secret:Gather available email ids from target domain"""
+
+    opts = {
+        'client_id': '',
+        'client_secret': ''
+    }
+
+    # Option descriptions. Delete any options not applicable to this module.
+    optdescs = {
+        'client_id': "Client ID for snov.io API",
+        'client_secret': "Client Secret for snov.io API"
+    }
+
+    results = None
+    errorState = False  
+
+    # More than 100 per response is not supported by Snov API
+    limit = 100
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+        
+
+    # What events is this module interested in for input
+    # For a list of all events, check sfdb.py.
+    def watchedEvents(self):
+        return ["DOMAIN_NAME", "INTERNET_NAME"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["EMAILADDR"]
+    
+    # Get Authentication token from Snov.IO API
+    def queryAccessToken(self):
+        params = {
+            'grant_type': "client_credentials",
+            'client_id': self.opts['client_id'],
+            'client_secret': self.opts['client_secret']
+        }
+
+        headers = {
+            'Accept': "application/json",
+        }
+        
+        res = self.sf.fetchUrl(
+            'https://api.snov.io/v1/oauth/access_token?' + urllib.parse.urlencode(params),
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        if not res['code'] == '200':
+            self.sf.error("Could not authenticate credentials", False)
+            self.errorState = True 
+            return None
+        try:
+            # Extract access token from response
+            content = res.get('content')
+            if isinstance(content, str):
+                accessToken = json.loads(content).get('access_token')
+            else:
+                accessToken = res.get('access_token')
+
```

Comment:
I'm not sure what the above `if` block is supposed to be doing? Won't the response always be a string?